src/main/java/com/zaxxer/hikari/HikariConfig.java

```diff
@@ -880,11 +882,17 @@ protected void loadProperties(String propertyFileName)
 
    private int generatePoolNumber()
    {
-      // Pool number is global to the VM to avoid overlapping pool numbers in classloader scoped environments
-      synchronized (System.getProperties()) {
-         final int next = Integer.getInteger("com.zaxxer.hikari.pool_number", 0) + 1;
-         System.setProperty("com.zaxxer.hikari.pool_number", String.valueOf(next));
-         return next;
+      try {
```

Comment:
Yes, it would probably be better to wrap this whole block into a doPrivileged(). I also don't like that if requires read access to *all* properties just for the purpose of synchronization, while the synchronization seems a little suspect in that I don't think there is a guarantee it will always be the same instance.